configure.ac

```diff
@@ -270,6 +270,27 @@ if test "x$RPCGEN" = "xno"; then
    AC_MSG_ERROR([`rpcgen` not found, glusterfs needs `rpcgen` exiting..])
 fi
 
+AC_CHECK_PROG([MOUNT], [mount], ["$(type mount | awk '{print $3}')"], [no])
```

Comment:
Wouldn't it be better to use `which` instead of `type` ?

Reply:
Not sure. On my system, `which` is a weird alias:
```
$ type which
which is a function
which () 
{ 
    ( alias;
    eval ${which_declare} ) | /usr/bin/which --tty-only --read-alias --read-functions --show-tilde --show-dot "$@"
}
```
I would prefer `type -p` which displays full path only, but `-p` is a `bash`-specific extension which is not available elsewhere (e.g. *BSD `sh`).

Reply:
`which` should work for normal cases and doesn't need the `awk` post-processing. The cases where `which` won't work, `type` won't work either. For example an alias:

```
$ which ls
alias ls='ls --color=auto'
        /usr/bin/ls
$ type ls
ls is aliased to `ls --color=auto'
```

`type` even doesn't show the path to the executable. `which` shows it, though it would need special handling in those cases. I'm not sure if we really need to deal with that. I would say a plain `which` should be enough, and AFAIK it's the standard way of locating executables.

Reply:
OK for `which`.